javascript/ql/src/Declarations/UnreachableMethodOverloads.ql

```diff
@@ -100,6 +100,13 @@ predicate signaturesMatch(MethodSignature method, MethodSignature other) {
   method.getName() = other.getName() and
   // same number of parameters.
   method.getBody().getNumParameter() = other.getBody().getNumParameter() and
+  // same this parameter (if exists)
+  (
+    not exists(method.getBody().getThisTypeAnnotation()) and
+    not exists(other.getBody().getThisTypeAnnotation())
+    or
+    method.getBody().getThisTypeAnnotation() = other.getBody().getThisTypeAnnotation()
```

Comment:
Did you mean to compare two AST nodes here? That should always be false, except when comparing a signature to itself.

Reply:
Yes, you are right, that would always fail if there existed a this-parameter.  
I now use `.getType()`, which will have no result for JavaScript, but since this is a TypeScript query that is not an issue.